File: crates/klyron_shell/bindings/cpp/builder.cpp
#include "builder.hpp"

namespace klyron::_shell {

ShellBuilder::ShellBuilder() : config_{} {}

ShellBuilder& ShellBuilder::set_config(const ShellConfig& cfg) {
    config_ = cfg;
    return *this;
}

ShellConfig ShellBuilder::config() const {
    return config_;
}

}

File: crates/klyron_shell/bindings/rust/builder.rs
//! Builder for klyron_shell
use crate::types::ShellConfig;

pub struct ShellBuilder {
    config: ShellConfig,
}

impl ShellBuilder {
    pub fn new() -> Self {
        Self { config: ShellConfig::default() }
    }
    pub fn with_version(mut self, version: &str) -> Self {
        self.config.version = version.to_string();
        self
    }
    pub fn build(self) -> ShellConfig {
        self.config
    }
}
